pkg/controller/hvpa/hvpa_controller.go

```diff
@@ -89,9 +147,126 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	eventHandler := &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: handler.ToRequestsFunc(func(a handler.MapObject) []reconcile.Request {
+			pod := a.Object.(*corev1.Pod)
+			nodeName := pod.Spec.NodeName
+			if nodeName == "" {
+				return nil
+			}
+			client := mgr.GetClient()
+			req := types.NamespacedName{
+				Name: nodeName,
+			}
+			if pod.Status.Reason == "Evicted" {
+				// If pod was evicted beause of 'KubeletHasInsufficientMemory' node condition,
+				// only then we want to continue, otherwise exit
+				node := corev1.Node{}
+				err := client.Get(context.TODO(), req, &node)
+				if err != nil {
+					log.Error(err, "Error fetching node")
+					return nil
+				}
+
+				hasMemPressure := false
+				for _, condition := range node.Status.Conditions {
+					if condition.Reason == "KubeletHasInsufficientMemory" && condition.Status == corev1.ConditionTrue {
+						hasMemPressure = true
+						break
+					}
+				}
+				if !hasMemPressure {
+					log.V(3).Info("Pod was evicted, but the node is not under memory pressure.")
+					return nil
+				}
+				log.V(3).Info("Pod might have been evited because node was under memory pressure")
+			}
+
+			owners := pod.GetOwnerReferences()
+			deploy := &appsv1.Deployment{}
+			for _, owner := range owners {
```

Comment:
This would be not only costly (in terms of the number of GET calls) but also restricted to only `Deployments`. Why not the reverse mapping by looping over HVPA resources especially if we also cache the selector in the `cachedNames` data structure?

Reply:
Changed the logic to use labels and selectors